spec/parsers/mp4_parser_spec.rb

```diff
@@ -31,6 +31,18 @@
       end
     end
 
+    context "when when magic bytes are present but the 'ftyp' box can't be parsed" do
+      let(:result) do
+        input = [0xFF].pack('N') + 'ftyp' + 'avc1 ' + [0x1].pack('N')
+        io = StringIO.new(input)
+        subject.call(io)
+      end
+
+      it 'should be nil' do
```

Comment:
I would keep the `subject` call in the test itself, and only prepare the "ingredients" for a test in a `let`, otherwise it feels like a scope creep between the two (`subject` should technically be the subject of a test, not of a _helper_).